File: sso-auth/frontend/features/connected-apps/model/connected-app.ts
export type ConnectedApp = {
  clientId: string
  name: string
  description: string
  callbackUrl: string
  enabled: boolean
  assertionTtlSeconds: number
}

export const defaultRelayApp = (): ConnectedApp => ({
  clientId: 'relay-agent',
  name: 'Masih Awam Relay',
  description: 'Agentic AI relay for Masih Awam.',
  callbackUrl: 'http://localhost:3100/connections/callback',
  enabled: true,
  assertionTtlSeconds: 90
})

File: sso-auth/src/domain/connected_app.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ConnectedApp {
    pub client_id: String,
    pub name: String,
    pub description: String,
    pub callback_url: String,
    pub enabled: bool,
    pub assertion_ttl_seconds: u64,
}
